Content.Client/Ghost/GhostSystem.cs

```diff
@@ -96,11 +96,18 @@ private void OnToggleFoV(EntityUid uid, EyeComponent component, ToggleFoVActionE
 
         private void OnToggleGhosts(EntityUid uid, GhostComponent component, ToggleGhostsActionEvent args)
         {
-            if (args.Handled)
+            if (args.Handled)  {
                 return;
+            }
 
-            Popup.PopupEntity(Loc.GetString("ghost-gui-toggle-ghost-visibility-popup"), args.Performer);
-
+            if (GhostVisibility == false)
+            {
+                Popup.PopupEntity(Loc.GetString("ghost-gui-toggle-ghost-visibility-popup-off"), args.Performer);
+            }
+            else
+            {
+                Popup.PopupEntity(Loc.GetString("ghost-gui-toggle-ghost-visibility-popup-on"), args.Performer);
+            }
```

Comment:
Did you test this? It looks like it's inverted since this is before `ToggleGhostVisibility()`.

Assuming it's correct, change it like this to get rid of the code dupe:
```suggestion
            var locId = GhostVisibility
                ? "ghost-gui-toggle-ghost-visibility-popup-on"
                : "ghost-gui-toggle-ghost-visibility-popup-off";
            Popup.PopupEntity(Loc.GetString(locId), args.Performer);
```